src/functionalTest/java/uk/gov/hmcts/reform/pip/account/management/controllers/TestingSupportApiTest.java

```diff
@@ -102,11 +116,146 @@ class TestingSupportApiTest {
     private UserRequest userRequest;
 
+    @Autowired
+    UserCollectionRequestBuilder userCollectionRequestBuilder;
+
+    @Autowired
+    UserCollectionRequest userCollectionRequest;
+
+    @Autowired
+    GraphServiceException graphServiceException;
+
+    @Mock
+    private UserCollectionPage userCollectionPage;
+
+    @Mock
+    private ClientConfiguration clientConfiguration;
+
     @BeforeAll
     static void startup() {
         OBJECT_MAPPER.findAndRegisterModules();
     }
 
+    @Test
+    void testTestingSupportCreateAccount() throws Exception {
+
+        //Azure mock setup
+        User userToReturn = new User();
+        userToReturn.id = ID;
+        userToReturn.givenName = GIVEN_NAME;
+
+        when(graphClient.users()).thenReturn(userCollectionRequestBuilder);
+        when(userCollectionRequestBuilder.buildRequest()).thenReturn(userCollectionRequest);
+        when(userCollectionRequest.post(any())).thenReturn(userToReturn);
+
+        userCollectionPage = new UserCollectionPage(new ArrayList<>(), userCollectionRequestBuilder);
+
+        when(clientConfiguration.getB2cUrl()).thenReturn(B2C_URL);
+        when(graphClient.users()).thenReturn(userCollectionRequestBuilder);
+        when(userCollectionRequestBuilder.buildRequest()).thenReturn(userCollectionRequest);
+        when(userCollectionRequest.filter(any())).thenReturn(userCollectionRequest);
+        when(userCollectionRequest.get()).thenReturn(userCollectionPage);
+
+        //Create new test user account
+        AzureAccount newAccount = createAccount(PASSWORD);
+
+        MockHttpServletRequestBuilder mockHttpServletRequestBuilder = MockMvcRequestBuilders
+            .post(TESTING_SUPPORT_CREATE_ACCOUNT_URL)
+            .content(OBJECT_MAPPER.writeValueAsString(newAccount))
+            .header(ISSUER_HEADER, ISSUER_ID)
+            .contentType(MediaType.APPLICATION_JSON);
+
+        MvcResult postResponse = mockMvc.perform(mockHttpServletRequestBuilder)
+            .andExpect(status().isCreated())
+            .andReturn();
+
+        PiUser createdAccount = OBJECT_MAPPER.readValue(postResponse.getResponse().getContentAsString(), PiUser.class);
+
+        assertEquals(EMAIL, createdAccount.getEmail(),
+                     "Azure account creation error"
+        );
+
+        //Reset azure mock setup
+        Mockito.reset(graphClient, userCollectionRequest, userCollectionRequestBuilder);
+
+        //User mock setup
+        when(graphClient.users(any())).thenReturn(userRequestBuilder);
+        when(userRequestBuilder.buildRequest()).thenReturn(userRequest);
+        when(userRequest.get()).thenReturn(new User());
+
+        //Check whether account is created in Pi user table
+        MockHttpServletRequestBuilder getPiUserRequest = get(ACCOUNT_URL + createdAccount.getUserId());
+
+        MvcResult responseGetUser = mockMvc.perform(getPiUserRequest).andExpect(status().isOk()).andReturn();
+
+        PiUser returnedUser = OBJECT_MAPPER.readValue(
```

Comment:
Think lines 191 to 195 are redundant. On line 189, you are passing the user ID so obviously the returned user will have the same ID. Testing the get endpoint returns OK on line 189 is sufficient.